File: backend/app/routers/stats.py
from fastapi import APIRouter, Query
from app.models import (
    SummaryResponse, MonthlyStats, WeeklyStats,
    YearlyStats, PaceTrendItem, PersonalBestsResponse
)
from app.services.stats_service import (
    get_summary, get_monthly_stats, get_weekly_stats,
    get_yearly_stats, get_pace_trend, get_personal_bests
)

router = APIRouter(prefix="/api/stats", tags=["统计"])


@router.get("/summary", response_model=SummaryResponse)
def summary():
    return get_summary()


@router.get("/monthly", response_model=list[MonthlyStats])
def monthly(year: int = Query(...)):
    return get_monthly_stats(year)


@router.get("/weekly", response_model=list[WeeklyStats])
def weekly(year: int = Query(...), month: int = Query(...)):
    return get_weekly_stats(year, month)


@router.get("/yearly", response_model=list[YearlyStats])
def yearly():
    return get_yearly_stats()


@router.get("/pace-trend", response_model=list[PaceTrendItem])
def pace_trend(period: str = Query("monthly", enum=["monthly", "weekly", "yearly"])):
    return get_pace_trend(period)


@router.get("/personal-bests", response_model=PersonalBestsResponse)
def personal_bests():
    return {"personal_bests": get_personal_bests()}

File: frontend/src/stores/stats.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import type { StatsSummary, PeriodStats, PersonalBestsResponse, TrainingTypeStatsResponse } from '@/types'
import * as api from '@/utils/api'

export const useStatsStore = defineStore('stats', () => {
  const summary = ref<StatsSummary | null>(null)
  const monthlyStats = ref<PeriodStats[]>([])
  const weeklyStats = ref<PeriodStats[]>([])
  const yearlyStats = ref<PeriodStats[]>([])
  const paceTrend = ref<PeriodStats[]>([])
  const personalBests = ref<PersonalBestsResponse | null>(null)
  const trainingTypeStats = ref<TrainingTypeStatsResponse | null>(null)
  const loading = ref(false)

  async function fetchSummary() {
    loading.value = true
    try {
      summary.value = await api.getSummary()
    } finally {
      loading.value = false
    }
  }

  async function fetchMonthlyStats(year: number) {
    monthlyStats.value = await api.getMonthlyStats(year)
  }

  async function fetchWeeklyStats(year: number, month: number) {
    weeklyStats.value = await api.getWeeklyStats(year, month)
  }

  async function fetchYearlyStats() {
    yearlyStats.value = await api.getYearlyStats()
  }

  async function fetchPaceTrend(period: string = 'monthly') {
    paceTrend.value = await api.getPaceTrend(period)
  }

  async function fetchPersonalBests() {
    personalBests.value = await api.getPersonalBests()
  }

  async function fetchTrainingTypeStats() {
    trainingTypeStats.value = await api.getTrainingTypeStats()
  }

  return {
    summary, monthlyStats, weeklyStats, yearlyStats, paceTrend, personalBests, trainingTypeStats, loading,
    fetchSummary, fetchMonthlyStats, fetchWeeklyStats, fetchYearlyStats, fetchPaceTrend, fetchPersonalBests, fetchTrainingTypeStats
  }
})
